packit/utils/obs_helper.py

```diff
@@ -0,0 +1,453 @@
+    (descr_elem := ET.Element("description")).text = description or ""
+    (person_elem := ET.Element("person")).attrib["userid"] = owner
+    person_elem.attrib["role"] = "maintainer"
+
+    for elem in (title_elem, descr_elem, person_elem):
+        root.append(elem)
+
+    for repo in repos:
+        (repo_elem := ET.Element("repository")).attrib["name"] = repo.name
+
+        for path_entry in repo.path:
+            (path_elem := ET.Element("path")).attrib["project"] = path_entry.project
+            path_elem.attrib["repository"] = path_entry.repository
+            repo_elem.append(path_elem)
+
+        for arch in repo.arch:
+            (arch_elem := ET.Element("arch")).text = arch
+            repo_elem.append(arch_elem)
+
+        root.append(repo_elem)
+
+    return root
+
+
+def parse_changelog_subject(line: str):
+    date: list[str]
+    author: list[str] = []
+    email: str = ""
+    version: str = ""
+
+    toks = [tok.strip() for tok in line.split(" ") if tok.strip()]
+
+    toks.pop(0)  # Remove leading "*"
+    date = toks[:4]
+
+    del toks[:4]
+    for index, element in enumerate(toks):
+        if not element.startswith("<"):
+            author.append(element)
+        else:
+            email += element
+            del toks[: index + 1]
+            break
+
+    if toks and toks[0] == "-":
+        toks.pop(0)
+
+    if toks:
+        version = toks[0]
+
+    return date, author, email, version
+
+
+def format_changelog(changelog_str: str) -> str:
+    """Converts spec changelog to OBS .changes format"""
+    output: list[str] = []
+    changelog_time = "12:00:00 UTC"
+    changelog = changelog_str.splitlines()
+
+    version_pattern = re.compile(r"-\s[0-9]+[\.\-]*")  # Precompile regex
+
+    for line in changelog:
+        if not line.strip():
+            continue
+        if line.strip().startswith("*"):
+            if output:
+                output[-1] += "\n"
+
+            date, author, email, version = parse_changelog_subject(line)
+            date.insert(3, changelog_time)
+
+            output.append("-" * 68)  # Set demarcator
+            output.append(f"{' '.join(date)} - {' '.join(author)} {email}\n")
+            if version:
+                output.append(f"- {version}")
+        else:
+            if version_pattern.match(line):  # Match version line
+                output.append(line)
+            else:
+                output.append(f"  {line}")
+
+    return "\n".join(output)
+
+
+def create_changes_file(package_dir: Path) -> None:
+    """
+    Creates a changelog file from a by copying the changelog section from the spec file.
+    Sets release to 0 since obs handles release numbers
+
+    Args:
+        package_dir (Path): Path to the directory containing the package spec file.
+
+    Returns:
+        None
+    """
+    specfile: str = ""
+    content: str
+    release_line = "Release:    0"
+
+    for file in list(filter(os.path.isfile, os.listdir(package_dir))):
```

Comment:
```suggestion
    for file in package_dir.iterdir():
        if file.is_file():
```

this looks a bit more readable to me